horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -570,17 +574,53 @@ Response addRunFromData(String start, String stop, String test,
       DatastoreResponse response = datastore.handleRun(data, metadata, testEntity.backendConfig, Optional.ofNullable(schemaUri), mapper);
 
       List<Integer> runIds = new ArrayList<>();
-      if (datastore.uploadType() == Datastore.UploadType.MUILTI && response.payload instanceof  ArrayNode){
-         response.payload.forEach(jsonNode -> {
-            runIds.add(getPersistRun(start, stop, test, owner, access, token, schemaUri, description, null, jsonNode, testEntity));
-         });
+      String responseString;
+      if (datastore.uploadType() == Datastore.UploadType.MUILTI
+              && response.payload instanceof ArrayNode ){
+
+         //if we return more than 10 results, offload to async queue to process - this might take a LOOONG time
+         if ( response.payload.size() > 10 ) {
+            response.payload.forEach(jsonNode -> {
+               mediator.queueRunUpload(start, stop, test, owner, access, token, schemaUri, description, null, jsonNode, testEntity);
+            });
+         } else { //process synchronously
+            response.payload.forEach(jsonNode -> {
+               runIds.add(getPersistRun(start, stop, test, owner, access, token, schemaUri, description, metadata, response.payload, testEntity));
+            });
+         }
       } else {
          runIds.add(getPersistRun(start, stop, test, owner, access, token, schemaUri, description, metadata, response.payload, testEntity));
       }
-//      return Response.status(Response.Status.OK).entity(String.valueOf(runId)).header(HttpHeaders.LOCATION, "/run/" + runId).build();
-      String reponseString = String.valueOf(runIds.stream().map(val -> Integer.toString(val)).collect(Collectors.joining(", ")));
-      return Response.status(Response.Status.OK).entity(reponseString).build();
+
+      responseString = runIds.size() > 0 ?
+              String.valueOf(runIds.stream().map(val -> Integer.toString(val)).collect(Collectors.joining(", ")))
+              : "More than 10 runs uploaded, processing asynchronously";
+
+      return Response.status(Response.Status.OK).entity(responseString).build();
```

Comment:
Can we return different status codes in according to whether the process has been sync or async?
E.g., `200` is the result contains the list of runIds and `202` is the results does not contain anything useful as it is going to be processed async, wdyt?

Reply:
Yes, 202 makes sense in this case.  I will change the return code and update the test